### PR TITLE
Provided.al2023 implementation (#139)

### PR DESCRIPTION
**Pull request by hawflau**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: manifest-v2.json
Comment (mndeveci) [inline manifest-v2.json:1]: I think `al2023/go/hello-img` is missing here?

**Review by moelasmar — APPROVED** [review]

File: tests/integration/unit_test/test_unit_test_go1_x.py [hunk 1]
```diff
@@ -10,6 +10,10 @@ class UnitTest_provided_go_cookiecutter_aws_sam_hello_golang(UnitTestBase.GoUnit
     directory = "al2/go/hello"
     code_directories = ["hello-world"]
 
+class UnitTest_providedal2023_go_cookiecutter_aws_sam_hello_golang(UnitTestBase.GoUnitTestBase):
```
Comment (moelasmar) [inline tests/integration/unit_test/test_unit_test_go1_x.py:13]: we need to add unit test case for the go/hello-img

File: al2023/rust/hello/cookiecutter.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+  "project_name": "My Project",
+  "project_slug": "{{ cookiecutter.project_name.lower().replace(' ', '-') }}",
+  "architectures": {
+    "value": []
```
Comment (moelasmar) [inline al2023/rust/hello/cookiecutter.json:5]: I think we need to set "x86_64" as one of the accepted values

File: tests/integration/unit_test/test_unit_test_go1_x.py (hunk 1 above)
Comment (hawflau) [reply tests/integration/unit_test/test_unit_test_go1_x.py:13]: it seems like we don't do unit test for hello-img for any runtime. We only do the "build_invoke" test for hello-img for all runtimes

Comment (moelasmar) [reply tests/integration/unit_test/test_unit_test_go1_x.py:13]: also it seems that sam init is failing for this template, not sure why

Comment (hawflau) [reply tests/integration/unit_test/test_unit_test_go1_x.py:13]: the cookiecutter.json seems to contain incorrect values. Fixed it and now waiting for the test run

**Review by mndeveci — APPROVED** [review]

**Pull request merged**